osu.Game/Overlays/BeatmapSet/LeaderboardModSelector.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics;
+using osu.Game.Rulesets.Mods;
+using osu.Framework.Bindables;
+using osu.Game.Rulesets;
+using osuTK;
+using osu.Game.Rulesets.UI;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics.UserInterface;
+using osuTK.Graphics;
+using System;
+using System.Linq;
+using osu.Framework.Extensions.IEnumerableExtensions;
+
+namespace osu.Game.Overlays.BeatmapSet
+{
+    public class LeaderboardModSelector : CompositeDrawable
+    {
+        public readonly BindableList<Mod> SelectedMods = new BindableList<Mod>();
+
+        private RulesetInfo ruleset;
+
+        public RulesetInfo Ruleset
+        {
+            get => ruleset;
+            set
+            {
+                if (ruleset?.Equals(value) ?? false)
+                {
+                    DeselectAll();
```

Comment:
Why is this a thing? Feels like a very hacky way of resetting the display.

Reply:
So is it better to make it a bindable and move resetting logic outside?

Reply:
Probably? Setting something to the same value generally shouldn't change the display/state unless this is obvious / well documented.